torchgeo/datasets/inria.py

```diff
@@ -0,0 +1,243 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Inria Aerial Image Labeling Dataset."""
+
+import glob
+import os
+from typing import Any, Callable, Dict, List, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio as rio
+import torch
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from torchgeo.datasets.geo import VisionDataset
+from torchgeo.datasets.utils import (
+    check_integrity,
+    extract_archive,
+    percentile_normalization,
+)
+
+
+class InriaBuildings(VisionDataset):
+    r"""Inria Aerial Image Labeling Dataset.
+
+    The `Inria Aerial Image Labeling
+    <https://project.inria.fr/aerialimagelabeling/>`_ dataset is a building
+    detection dataset over dissimilar settlements ranging ranging from densely
+    populated areas to alpine towns.
+
+    Dataset features:
+
+    * Coverage of 810 km\ :sup:`2`\  (405 km\ :sup:`2`\  for training and 405
+      km\ :sup:`2`\  for testing)
+    * Aerial orthorectified color imagery with a spatial resolution of 0.3 m
+    * Train cities: Austin, Chicago, Kitsap, West Tyrol, Vienna
+    * Test cities: Bellingham, Bloomington, Innsbruck, San Francisco, East Tyrol
+
+    Dataset format:
+
+    * Imagery - Aerial GeoTIFFs
+    * Labels - Aerial GeoTIFFs
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://doi.org/10.1109/IGARSS.2017.8127684
+
+
+    .. versionadded:: 0.3
+    """
+
+    foldername = "AerialImageDataset"
```

Comment:
This is called `directory` in other datasets. Not sure if we should standardize though.